File: COURS/C/C-09/nested_if.c
#include <stdio.h>    // For printf
#include <stdbool.h>  // To use boolean expressions

// Nested if-else statement in c.
int main()
{
    float price = 10.00;         // Original price of the ticket
    bool isStudent = true;       // Is the user a student? (10% discount if true)
    bool isSenior = true;        // Is the user a senior? (20% discount if true)

    // Check if the user is a student
    if (isStudent)
    {
        // If the user is also a senior
        if (isSenior)
        {
            printf("You got a student discount of 10%%\n");
            printf("You got a senior discount of 20%%\n");
            price *= 0.7;  // Apply combined discount: 10% + 20% = 30%
        }
        else
        {
            printf("You got a student discount of 10%%\n");
            price *= 0.9;  // Apply only student discount
        }
    }
    else
    {
        // If not a student, check if the user is a senior
        if (isSenior)
        {
            printf("You got a senior discount of 20%%\n");
            price *= 0.8;  // Apply only senior discount
        }
        // Else: no discounts applied
    }

    // Show final price after discounts
    printf("The price of the ticket is: $%.2f\n", price);

    return 0;
}

File: COURS/JAVA/JAVA-09/NestedIf.java
public class NestedIf {
    public static void main(String[] args) {

        // Initialize customer information
        boolean isStudent = true; // true if the person is a student
        boolean isSenior = true;  // true if the person is a senior (e.g. age 60+)
        double price = 9.99;      // base price of the ticket

        // Check if the person is a student
        if (isStudent) {
            // Inside student block, check if also a senior
            if (isSenior) {
                // Both student and senior → apply 10% + 20% = 30% total discount
                System.out.println("You get a senior discount of 20%");
                System.out.println("You get a student discount of 10%");
                price *= 0.7; // Apply 30% discount
            } else {
                // Only student → apply 10% discount
                System.out.println("You get a student discount of 10%");
                price *= 0.9;
            }
        } else {
            // Not a student
            if (isSenior) {
                // Only senior → apply 20% discount
                System.out.println("You get a senior discount of 20%");
                price *= 0.8;
            } else {
                // Not student, not senior → no discount
                price *= 1; // No change
            }
        }

        // Print the final price formatted to 2 decimal places
        System.out.printf("The price of a ticket is: $%.2f", price);
    }
}
